Sources/PubNubChatComponents/Message List/MessageListViewController.swift

```diff
@@ -133,4 +206,168 @@ open class ChatViewController: CollectionViewComponent {
   fileprivate var isUserInitiatedScrolling: Bool {
     return collectionView.isDragging || collectionView.isDecelerating
   }
+  
+  // MARK: UIViewControllerTransitioningDelegate
+
+  @objc(presentationControllerForPresentedViewController:presentingViewController:sourceViewController:)
+  public func presentationController(
+    forPresented presented: UIViewController,
+    presenting: UIViewController?,
+    source: UIViewController
+  ) -> UIPresentationController? {
+    guard let gestureRecognizedCell = gestureRecognizedCell else {
+      return nil
+    }
+    return AddMessageReaction.PresentationController(
+      presentedViewController: presented,
+      presentingViewController: presenting,
+      messageCell: gestureRecognizedCell,
+      messageReactionsComponentSize: preferredPickerViewSize
+    )
+  }
+  
+  @objc(animationControllerForDismissedController:)
+  public func animationController(forDismissed dismissed: UIViewController) -> UIViewControllerAnimatedTransitioning? {
+    AddMessageReaction.AnimationController(duration: 0.15, isBeingPresented: false)
+  }
+  
+  @objc(animationControllerForPresentedController:presentingController:sourceController:)
+  public func animationController(forPresented presented: UIViewController, presenting: UIViewController, source: UIViewController) -> UIViewControllerAnimatedTransitioning? {
+    AddMessageReaction.AnimationController(duration: 0.25, isBeingPresented: true)
+  }
+}
+
+// MARK: Showing message reactions
+
+private enum AddMessageReaction {
```

Comment:
Is this enum just for namespacing? If it's private why do we need to namespace?

Reply:
Yes, that's for namespacing only. I wanted to somehow group all classes that are responsible for showing a picker view. Do you suggest to remove the `private` modifier or would you like to completely get rid of this enum?

Reply:
No need to change anything, I just wanted to make sure I was understanding its purpose.  As I mentioned in the other comment, I'm assuming that eventually we'll need to move away from hardcoded values so that might require use to re-architect the solution so need to make unnecessary changes right now.  

The only optional change that you could do is move it to its own file and just change it be `internal` instead of `private`, or instead make it an extension of the `MessageListViewController` since they're so tightly coupled already.    Neither of those are very necessary